src/fastoad/gui/variable_viewer.py

```diff
@@ -423,7 +425,7 @@ def _filter_variables(
             else:
                 path = ":".join(modules)
 
-        path_filter = [True] * len(df) if path == "" else df.Name.str.startswith(path)
+        path_filter = [True] * len(df) if path == ("" or TAG_ALL) else df.Name.str.startswith(path)
```

Comment:
At first, you made me think that if `path == ("" or TAG_ALL)` could be a Python feature I did not know about. But in my tests, it does not work: `"" or TAG_ALL` is actually `TAG_ALL` so your test is equivalent to `path == TAG_ALL`.
If you want to test that, you need to write `path in ["", TAG_ALL]`. But if you do so, then the line `filtered_df = df[path_filter & io_filter]` below may fail because `&` does not work if given 2 lists (`[True] * len(df)`) 

And it looks like your code does not encounter this error because your `if` statement is always `False` so that the line looks equivalent to `path_filter = df.Name.str.startswith(path)`.

Therefore, I have these 2 questions:
 - When is `path` supposed to be equal to `TAG_ALL`?
 - Can the line `path_filter = df.Name.str.startswith(path)` be enough?

Reply:
Thanks indeed I think I got mixed up with other programming languages ...
Yes I guess `path` always contains `TAG_ALL` now.
Your proposition works and seems enough for our usage.
Corrected in fe4c339.

Reply:
Thanks. Actually, in my tests, `path` never contains `TAG_ALL` 😅 
Yet, `path` can be empty, but it is not a problem here.